tests/meta/workflow/test_analysis.py

```diff
@@ -63,3 +69,290 @@ def test_gene_panels_not_added(customer_id):
+    mip_analysis_api, analysis_store: Store, helpers
+):
+    """Tests that ensure_flow_cells_on_disk does not perform any action
+    when is_flow_cell_check_applicable returns false."""
+
+    # GIVEN a case with a REMOVED flow cell
+    case: Family = analysis_store.get_cases()[0]
+    helpers.add_flowcell(
+        analysis_store,
+        flow_cell_name="flow_cell_test",
+        archived_at=datetime.now(),
+        sequencer_type=Sequencers.NOVASEQ,
+        samples=analysis_store.get_samples_by_case_id(case.internal_id),
+        status=FlowCellStatus.REMOVED,
+        date=datetime.now(),
+    )
+
+    # WHEN _is_flow_cell_check_available returns True
+    with mock.patch.object(
+        AnalysisAPI,
+        "_is_flow_cell_check_applicable",
+        return_value=True,
+    ):
+        mip_analysis_api.ensure_flow_cells_on_disk(case.internal_id)
+
+    # THEN flow cells should be requested for the case
+    assert analysis_store.get_flow_cell_by_name("flow_cell_test").status == FlowCellStatus.REQUESTED
+
+
+def test_is_case_ready_for_analysis_true(mip_analysis_api, analysis_store, helpers):
+    """Tests that is_case_ready_for_analysis returns true for a case whose flow cells are all ON_DISK and whose
+    files need no decompression nor are being decompressed currently."""
+
+    case: Family = analysis_store.get_cases()[0]
+    helpers.add_flowcell(
+        analysis_store,
+        flow_cell_name="flowcell_test",
+        archived_at=datetime.now(),
+        sequencer_type=Sequencers.NOVASEQ,
+        samples=analysis_store.get_samples_by_case_id(case.internal_id),
+        status=FlowCellStatus.ON_DISK,
+        date=datetime.now(),
+    )
+    with mock.patch.object(
+        PrepareFastqAPI, "is_spring_decompression_needed", return_value=False
+    ), mock.patch.object(PrepareFastqAPI, "is_spring_decompression_running", return_value=False):
+        assert mip_analysis_api.is_case_ready_for_analysis(case_id=case.internal_id)
+
+
+def test_is_case_ready_for_analysis_decompression_needed(mip_analysis_api, analysis_store):
+    """Tests that is_case_ready_for_analysis returns true for a case whose flow cells are all ON_DISK and whose
+    files need no decompression nor are being decompressed currently."""
+
+    case: Family = analysis_store.get_cases()[0]
+    with mock.patch.object(
+        PrepareFastqAPI, "is_spring_decompression_needed", return_value=True
+    ), mock.patch.object(PrepareFastqAPI, "is_spring_decompression_running", return_value=False):
+        assert not mip_analysis_api.is_case_ready_for_analysis(case_id=case.internal_id)
+
+
+def test_is_case_ready_for_analysis_decompression_running(
+    mip_hk_store,
+    analysis_store,
+    helpers,
+    mip_analysis_api,
+):
+    """Tests that is_case_ready_for_analysis returns true for a case whose flow cells are all ON_DISK and whose
+    files need no decompression nor are being decompressed currently."""
+
```

Comment:
use GIVEN, WHEN, THEN for better readability

Reply:
Done!